pkg/controller/scyllacluster/sync_certs.go

```diff
@@ -0,0 +1,203 @@
+package scyllacluster
+
+import (
+	"context"
+	"crypto/x509/pkix"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+	"time"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	ocrypto "github.com/scylladb/scylla-operator/pkg/crypto"
+	"github.com/scylladb/scylla-operator/pkg/helpers"
+	"github.com/scylladb/scylla-operator/pkg/internalapi"
+	okubecrypto "github.com/scylladb/scylla-operator/pkg/kubecrypto"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apierrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+func (scc *Controller) syncCerts(
+	ctx context.Context,
+	sc *scyllav1.ScyllaCluster,
+	secrets map[string]*corev1.Secret,
+	configMaps map[string]*corev1.ConfigMap,
+	serviceMap map[string]*corev1.Service,
+) ([]metav1.Condition, error) {
+	// Be careful to always apply signers first to be sure they have been persisted
+	// before applying any child certificate signed by it.
+
+	var errors []error
+	var progressingConditions []metav1.Condition
+
+	cm := okubecrypto.NewCertificateManager(
+		scc.kubeClient.CoreV1(),
+		scc.secretLister,
+		scc.kubeClient.CoreV1(),
+		scc.configMapLister,
+		scc.eventRecorder,
+	)
+
+	clusterLabels := naming.ClusterLabels(sc)
+
+	// Manage client certificates.
+	errors = append(errors, cm.ManageCertificates(
+		ctx,
+		time.Now,
+		&sc.ObjectMeta,
+		scyllaClusterControllerGVK,
+		&okubecrypto.CAConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   naming.GetScyllaClusterLocalClientCAName(sc.Name),
+				Labels: clusterLabels,
+			},
+			Validity: 10 * 365 * 24 * time.Hour,
+			Refresh:  8 * 365 * 24 * time.Hour,
+		},
+		&okubecrypto.CABundleConfig{
+			MetaConfig: okubecrypto.MetaConfig{
+				Name:   naming.GetScyllaClusterLocalClientCAName(sc.Name),
+				Labels: clusterLabels,
+			},
+		},
+		[]*okubecrypto.CertificateConfig{
+			{
+				MetaConfig: okubecrypto.MetaConfig{
+					Name:   naming.GetScyllaClusterLocalUserAdminCertName(sc.Name),
+					Labels: clusterLabels,
+				},
+				Validity: 30 * 24 * time.Hour,
```

Comment:
This should be a long trusted certificate. We don't want to force our users to download new bundle every 30 days.

Reply:
correct, thx